### PR TITLE
Fix upgrade v2.2.1 -> v3.4.0 forces replacement 

### PR DESCRIPTION
**Pull request by bendbennett**
References: #302 

This should address https://github.com/hashicorp/terraform-provider-random/issues/302 insofar as it fixes the issue when using the [config](https://github.com/hashicorp/terraform-provider-random/issues/302#issue-1357367589) supplied by [blgm](https://github.com/blgm) and the [config](https://github.com/hashicorp/terraform-provider-random/issues/302#issuecomment-1233045227) supplied by [alexpoto](https://github.com/alexpoto).

### CODE REVIEW

File: internal/provider/resource_password_test.go [hunk 1]
```diff
@@ -506,6 +506,164 @@ func TestAccResourcePassword_Min(t *testing.T) {
 	})
 }
 
+// TestAccResourcePassword_UpgradeFromVersion2_2_1 requires that you are running an amd64 Terraform binary
+// if you are running this test locally on arm64 architecture otherwise you will see the following error:
+//
+// Error: Incompatible provider version
+//
+// Provider registry.terraform.io/hashicorp/random v2.2.1 does not have a
+// package available for your current platform ...
```
Comment (bflad) [inline internal/provider/resource_password_test.go:515]: I think we can codify this logic into the test via:

```go
TestCase{
  PreCheck: func() {
    if runtime.GOOS == "darwin" && runtime.GOARCH == "arm64" {
      t.Skip("This test requires darwin/amd64 to download the old provider version. Setting TF_ACC_TERRAFORM_PATH to darwin/amd64 compatible Terraform binary can be used as a workaround.")
    }
  },
  // ... other fields
}
```

Comment (bendbennett) [reply internal/provider/resource_password_test.go]: Thanks. Have incorporated.

**Review by bflad — APPROVED** [review]
Looks good to me 🚀

**Comment by blgm** [comment]
Thank you @bendbennett and @bflad!

**Pull request merged**